concrete/controllers/single_page/register.php

```diff
@@ -55,23 +55,23 @@ public function do_register()
 
             // clean the username
             $username = trim($username);
-            $username = preg_replace("/ +/", " ", $username);
+            $username = preg_replace('/ +/', ' ', $username);
 
-            if ($ip->isBanned()) {
+            if ($ip->isBlacklisted()) {
                 $e->add($ip->getErrorMessage());
             }
 
             if ($config->get('concrete.user.registration.captcha')) {
                 $captcha = $this->app->make('helper/validation/captcha');
                 if (!$captcha->check()) {
-                    $e->add(t("Incorrect image validation code. Please check the image and re-enter the letters or numbers as necessary."));
+                    $e->add(t('Incorrect image validation code. Please check the image and re-enter the letters or numbers as necessary.'));
                 }
             }
 
             if (!$vals->email($_POST['uEmail'])) {
                 $e->add(t('Invalid email address provided.'));
             } elseif (!$valc->isUniqueEmail($_POST['uEmail'])) {
-                $e->add(t("The email address %s is already in use. Please choose another.", $_POST['uEmail']));
+                $e->add(t('The email address %s is already in use. Please choose another.', $_POST['uEmail']));
```

Comment:
@Mnkras I don't know if a valid email address can contain code that could be used for XSS...